bitcoin/examples/bip32.rs

```diff
@@ -47,9 +47,11 @@ fn main() {
     println!("Public key at {}: {}", path, xpub);
 
     // generate first receiving address at m/0/0
-    // manually creating indexes this time
-    let zero = ChildNumber::ZERO_NORMAL;
-    let public_key = xpub.derive_pub(&secp, &[zero, zero]).unwrap().public_key;
+    let zero_path = DerivationPath::from_str("m/0/0").unwrap();
+    //While using only Normal Child Numbers, you can use NormalDerivationPath scruct converting
+    //from DerivationPath
+    let normal_zero_path = zero_path.try_into_normal().unwrap();
+    let public_key = xpub.derive_pub(&secp, normal_zero_path).public_key;
```

Comment:
I think that this example cannot change anymore.
For me that's what the ideal would be.

Reply:
Actually a better example would be parsing `NormalDerivationPath` directly.

Reply:
Like 
```Rust
    let zero_path = NormalDerivationPath::from_str("m/0/0").unwrap();
```
?

Reply:
Yes. Though `let zero_path = "0/0".parse::<NormalDerivationPath>().unwrap()` is considered idiomatic.

Reply:
Ok, so we're considering to implement the marjority of the `DerivationPath` functions to `NormalDerivationPath` ? This kinda runs out of my original idea of just using it as a filter and this being a small PR with  "low bug risk". But i agree that being able to work directly with `NormalDerivationPath` is convenient.

Reply:
I believe we should. It could be done one PR at a time but the issue is if we release an incomplete API it'll just confuse users.